src/WinGetServer/WinMain.cpp

```diff
@@ -18,27 +26,140 @@ static void _releaseNotifier() noexcept
     _comServerExitEvent.SetEvent();
 }
 
-int __stdcall wWinMain(_In_ HINSTANCE, _In_opt_ HINSTANCE, _In_ LPWSTR, _In_ int)
+unsigned char* GetUCharString(const char* in)
+{
+    return reinterpret_cast<unsigned char*>(const_cast<char*>(in));
+}
+
+HRESULT WindowsPackageManagerServerInitializeRPCServer()
+{
+    RPC_STATUS status = RpcServerUseProtseqEpA(GetUCharString("ncacn_np"), RPC_C_PROTSEQ_MAX_REQS_DEFAULT, GetUCharString("\\pipe\\WinGetServerManualActivation_SID"), nullptr);
+    RETURN_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+
+    wil::unique_hlocal_security_descriptor securityDescriptor;
+    RETURN_LAST_ERROR_IF(!ConvertStringSecurityDescriptorToSecurityDescriptorA("D:(A;;GA;;;BA)", SDDL_REVISION_1, &securityDescriptor, nullptr));

```

Comment:
This lets any admin connect; we only want to allow the user that this server is for to connect.

There is a principal for "self", `PS`.  Then one would also have to add the mandatory high label (I think) to require admin.